From ade546433d8650c28ab27be2bf3ca6e39b974d46 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 8 Jan 2021 21:09:48 +0100
Subject: [PATCH 112/121] dt-bindings: interrupt-controller: Document the
 Lantiq ICU

Many Lantiq SoCs with MIPS cores embed a so-called "Interrupt Controller
Unit". This is an interrupt-controller to which most system peripherals
are connected. Add the documentation for this IP.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 .../interrupt-controller/lantiq,icu.yaml      | 81 +++++++++++++++++++
 1 file changed, 81 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/interrupt-controller/lantiq,icu.yaml

diff --git a/Documentation/devicetree/bindings/interrupt-controller/lantiq,icu.yaml b/Documentation/devicetree/bindings/interrupt-controller/lantiq,icu.yaml
new file mode 100644
index 000000000000..5e82a88e197b
--- /dev/null
+++ b/Documentation/devicetree/bindings/interrupt-controller/lantiq,icu.yaml
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/interrupt-controller/lantiq,icu.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: Lantiq ICU (Interrupt Controller Unit)
+
+maintainers:
+  - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+
+allOf:
+  - $ref: /schemas/interrupt-controller.yaml#
+
+description:
+  The Lantiq Falcon, Amazon-SE, Danube, xRX100, xRX200 and xRX300 SoCs
+  come with an "Interrupt Controller Unit" (ICU). This is is a 5 to
+  160 (5 * 32) multiplexer and connects to most peripherals on these
+  SoCs.
+
+properties:
+  compatible:
+    items:
+      - enum:
+          - lantiq,falcon-icu
+          - lantiq,ase-icu
+          - lantiq,danube-icu
+          - lantiq,xrx100-icu
+          - lantiq,xrx200-icu
+          - lantiq,xrx300-icu
+      - const: lantiq,icu
+
+  reg:
+    description: One set of registers per CPU core
+    minItems: 1
+    maxItems: 2
+
+  interrupts:
+    minItems: 5
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    const: 1
+
+  '#address-cells':
+    const: 1
+
+required:
+  - compatible
+  - reg
+  - interrupt-parent
+  - interrupts
+  - interrupt-controller
+  - '#interrupt-cells'
+  - '#address-cells'
+
+additionalProperties: false
+
+examples:
+  - |
+    interrupt-controller@80200 {
+      compatible = "lantiq,ase-icu", "lantiq,icu";
+      reg = <0x80200 0xc8>;
+      interrupt-parent = <&cpu_irqc>;
+      interrupts = <2>, <3>, <4>, <5>, <6>;
+      interrupt-controller;
+      #interrupt-cells = <1>;
+      #address-cells = <1>;
+    };
+  - |
+    interrupt-controller@80200 {
+      compatible = "lantiq,xrx200-icu", "lantiq,icu";
+      reg = <0x80200 0xc8>, <0x80300 0xc8>;
+      interrupt-parent = <&cpu_irqc>;
+      interrupts = <2>, <3>, <4>, <5>, <6>;
+      interrupt-controller;
+      #interrupt-cells = <1>;
+      #address-cells = <1>;
+    };
+...
-- 
2.47.2

